src/components/Rating.jsx

```diff
@@ -266,85 +254,88 @@ export default class Rating extends Component {
   }
 
   /**
-   * calculate tempRating and apply highlighting to the component when it is clicked, touch ends, enter or space key are hit
+   * calculate focusedValue and apply highlighting to the component when it is clicked, touch ends, enter or space key are hit
    */
   _changeComponent(pageX) {
     const wrapperNode = React.findDOMNode(this.refs.wrapper);
     const wrapperWidth = wrapperNode.getBoundingClientRect().width;
     const mouseMoved = pageX - wrapperNode.getBoundingClientRect().left;
-    const newRating = Math.round(mouseMoved * 5 / wrapperWidth + .4);
-    this._showTempRating(newRating);
+    const newRating = Math.round(mouseMoved * 5 / wrapperWidth + 0.4);
+    this._showFocusedValue(newRating);
   }
 
   /**
    * update component component is clicked, touch ends, enter or space key are hit.
    */
   _updateComponent() {
-    var ratingValue = this.state.tempRating > 0 ? this.state.tempRating : undefined;
-    if(this.props.valueLink) {
-      this.props.valueLink.requestChange(ratingValue);
-    }
-    else if(!this.props.value && this.props.defaultValue) {
-      this.setState({
-        rating: ratingValue
-      });
-    }
+    var value = this.state.focusedValue > 0 ? this.state.focusedValue : undefined;
+
     this.setState({
```

Comment:
Hey Nik,

I am confused about this line below setting value in state for all (valueLink, value and defaultValue). This the rating shown by component with colored stars will be immediately updated. But what we rather want for controlled components is that node.value should update and onChange called, but visibly value should be seen update in a controlled way.